.github/workflows/ci-backend.yml

```diff
@@ -111,28 +111,65 @@ jobs:
         needs: changes
         if: ${{ needs.changes.outputs.backend == 'true' && github.repository == 'PostHog/posthog' }}
 
-        name: Django tests – Py ${{ matrix.python-version }}
+        name: Django tests – Py ${{ matrix.python-version }} ${{ matrix.name }} (${{matrix.group}}/${{ matrix.total-concurrency }})
         runs-on: ubuntu-latest
 
         strategy:
             fail-fast: false
             matrix:
-                python-version: ['3.8.5', '3.9.0']
+                python-version: ['3.8.5']
+                ee: [true]
+                foss: [false]
+                saml: [false]
+                name: ['']
+                total-concurrency: [5]
+                group: [1, 2, 3, 4, 5]
+                include:
+                    # :TRICKY: Run FOSS tests in a separate container
+                    - python-version: '3.8.5'
+                      ee: false
+                      saml: false
+                      foss: true
+                      name: 'FOSS'
+                      total-concurrency: 1
+                      group: 1
+                    # :TRICKY: Run FOSS tests in a separate container
+                    - python-version: '3.9.0'
+                      ee: false
+                      saml: false
+                      foss: true
+                      name: 'FOSS'
+                      total-concurrency: 1
+                      group: 1
+                    # :TRICKY: Run SAML tests in a separate container
+                    - python-version: '3.8.5'
```

Comment:
I believe we're installing SAML deps on all installations now, so we could run SAML tests together with everything else. CC @guidoiaquinti

Reply:
We don't install SAML tests except for saml tests. the `if:` condition prevents it from running otherwise

Reply:
I know this is the case in this file, but I believe @guidoiaquinti was working on something so that SAML deps are always installed, and avoid having extra checks and complexity here.

Reply:
Unfortunately https://github.com/PostHog/posthog/pull/6607 didn't land due to issues with Heroku and CI (unfortunately we don't run CI and Heroku in containers and I think this is a blocker for my PR to land)